File: python/hot100/87.最长递增子序列.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        ng = 0
        for x in nums:
            j = bisect_left(nums, x, 0, ng)
            nums[j] = x
            if j == ng:
                ng += 1
        return ng

File: cpp/hot100/87.最长递增子序列.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        auto end = nums.begin();
        for(auto x : nums){
            auto it = lower_bound(nums.begin(), end, x);
            *it = x;
            if(it == end){
                ++end;
            }
        }
        return end - nums.begin();
    }
};